icu4c/source/test/cintltst/ulocaletst.c

```diff
@@ -0,0 +1,321 @@
+// © 2023 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+#include "unicode/ctest.h"
+#include "unicode/uloc.h"
+#include "unicode/ulocale.h"
+#include "cintltst.h"
+#include "cmemory.h"
+#include "cstring.h"
+
+#define WHERE __FILE__ ":" XLINE(__LINE__) " "
+#define XLINE(s) LINE(s)
+#define LINE(s) #s
+
+#define TESTCASE(name) addTest(root, &name, "tsutil/ulocaletst/" #name)
+
+enum {
+    LANGUAGE = 0,
+    SCRIPT,
+    REGION,
+    VAR,
+    NAME,
+    BASENAME,
+};
+static const char* const rawData[][9] = {
+    // language code
+    {   "en",   "fr",   "ca",   "el",   "no",   "it",   "xx",   "zh",
+         "x-klingon"  },
+    // script code
+    {   "",     "",     "",     "",     "",     "",     "",     "Hans",
+        "Latn"},
+    // country code
+    {   "US",   "FR",   "ES",   "GR",   "NO",   "",     "YY",   "CN",
+        "ZX"},
+    // variant code
+    {   "",     "",     "",     "",     "NY",   "",     "",   "",
+        ""},
+    // full name
+    {   "en_US",    "fr_FR",    "ca_ES",    "el_GR",    "no_NO_NY", "it",   "xx_YY",   "zh_Hans_CN",
+        "x-klingon_Latn_ZX.utf32be@special"},
+    // basename
+    {   "en_US",    "fr_FR",    "ca_ES",    "el_GR",    "no_NO_NY", "it",   "xx_YY",   "zh_Hans_CN",
+        "x-klingon_Latn_ZX.utf32be@special"},
+};
```

Comment:
I think this table would read a lot easier if you organized it the other way, where the individual fields were the columns and the individual test cases were the rows.  It also would mean you could add test cases to the table without having to change the dimension in line 25.